### PR TITLE
[ALLUXIO-2175] Deployment scripts for Alluxio with a Swift UFS

### PR DESCRIPTION
**Pull request by madanadit**
Modifications to vagrant scripts with Swift UFS configuration.

https://alluxio.atlassian.net/browse/ALLUXIO-2175

### CODE REVIEW

**Comment by peisun1115** [comment]
@madanadit Are you planning to add this to autobots as well so that we can do it in sandbox?

**Comment by uronce-cc** [comment]
Will more scripts be added for automatically downloading, configuring, and starting swift?

**Comment by madanadit** [comment]
@uronce-cc Yes, I will address that in a separate PR. Thanks.

**Comment by uronce-cc** [comment]
LGTM

File: deploy/vagrant/provision/roles/ufs_swift/files/config_alluxio.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+export ALLUXIO_UNDERFS_ADDRESS="swift://${SWIFT_CONTAINER}"
+
+export ALLUXIO_JAVA_OPTS+="
+  -Dfs.swift.user=${SWIFT_USER}
+  -Dfs.swift.tenant=${SWIFT_TENANT}
+  -Dfs.swift.password=${SWIFT_PASSWORD}
+  -Dfs.swift.auth.url=${SWIFT_AUTH_URL}
+  -Dfs.swift.use.public.url=${SWIFT_USE_PUBLIC_URL}
+  -Dfs.swift.auth.method=${SWIFT_AUTH_METHOD}
+"
```
Comment (calvinjia) [inline deploy/vagrant/provision/roles/ufs_swift/files/config_alluxio.sh]: Is it possible to use the `alluxio-site.properties` file to set this options? Though I guess the whole deploy module does this, so it can be refactored in another PR.

File: deploy/vagrant/provision/roles/ufs_swift/files/compile_alluxio.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env bash
+
+cd /alluxio
+mvn clean install -Dmaven.javadoc.skip=true -DskipTests -Dlicense.skip=true -Dcheckstyle.skip=true -Dfindbugs.skip=true -Pmesos > mvn.log 2>&1
```
Comment (jsimsa) [inline deploy/vagrant/provision/roles/ufs_swift/files/compile_alluxio.sh]: OOC, why is the mesos profile needed?

Comment (jsimsa) [inline deploy/vagrant/provision/roles/ufs_swift/files/compile_alluxio.sh]: answering my own question, I guess you copied that from elsewhere and it is used there because the deploy module supports Mesos

**Comment by jsimsa** [comment]
LGTM

File: deploy/vagrant/provision/roles/ufs_swift/files/config_alluxio.sh (hunk 1 above)
Comment (madanadit) [inline deploy/vagrant/provision/roles/ufs_swift/files/config_alluxio.sh]: @calvinjia Sounds good. I've created a ticket to address this for all UFSs: https://alluxio.atlassian.net/browse/ALLUXIO-2177

**Comment by calvinjia** [comment]
LGTM
